python/plugins/processing/core/ProcessingConfig.py

```diff
@@ -83,6 +84,10 @@ def initialize():
             ProcessingConfig.tr('Use only selected features'), True))
         ProcessingConfig.addSetting(Setting(
             ProcessingConfig.tr('General'),
+            ProcessingConfig.IGNORE_NULL_GEOMS,
+            ProcessingConfig.tr('Ignore features with null geometries'), True))
```

Comment:
I like you proposal. What I don't understand is what you say about worse performance. In your case, Processing also has to check for null geoms. The only difference isthat, instead of skipping them, it will signal them with that warning.

I will update this PR to include your proposal